flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -141,6 +140,25 @@ protected Connection setupConnection() throws Exception {
 		return setupConnectionFactory().newConnection();
 	}
 
+	/**
+	 * Initializes the consumer's {@link Channel}. If a prefetch count has been set in {@link RMQConnectionConfig},
+	 * the new channel will be use it for {@link Channel#basicQos(int)}.
+	 *
+	 * @param connection the consumer's {@link Connection}.
+	 * @return the channel.
+	 * @throws Exception if there is an issue creating or configuring the channel.
+	 */
+	protected Channel setupChannel(Connection connection) throws Exception {
```

Comment:
If I understand the RMQSource well, the reasons why `setupConnection` or `setupConnectionFactory` are protected is for assertions in tests. I see you managed to test it without using the `setupChannel` method. I think we can safely make it private.

Moreover WDYT of annotating the other methods with `VisibleForTesting`?

Reply:
I think both are great ideas, I'll implement those shortly.